src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1271,8 +1271,32 @@ private static int GetDeclarationCount(SyntaxNode declaration)
                 _ => 1,
             };
 
+        private static SyntaxNode EnsureRecordDeclarationHasBody(SyntaxNode declaration)
+        {
+            if (declaration is RecordDeclarationSyntax recordDeclaration)
+            {
+                if (recordDeclaration.SemicolonToken.IsKind(SyntaxKind.SemicolonToken))
+                {
+                    recordDeclaration = recordDeclaration.WithSemicolonToken(default);
+                }
+                if (!recordDeclaration.OpenBraceToken.IsKind(SyntaxKind.OpenBraceToken))
+                {
+                    recordDeclaration = recordDeclaration.WithOpenBraceToken(SyntaxFactory.Token(SyntaxKind.OpenBraceToken));
+                }
+                if (!recordDeclaration.CloseBraceToken.IsKind(SyntaxKind.CloseBraceToken))
+                {
+                    recordDeclaration = recordDeclaration.WithCloseBraceToken(SyntaxFactory.Token(SyntaxKind.CloseBraceToken));
+                }
+
+                return recordDeclaration;
+            }
+
+            return declaration;
+        }
+
         public override SyntaxNode InsertMembers(SyntaxNode declaration, int index, IEnumerable<SyntaxNode> members)
         {
+            declaration = EnsureRecordDeclarationHasBody(declaration);
```

Comment:
i'm not certain how this works with the code that exists below.  i.e. in the declaration switch lower on, we don't seem to handle records.

Reply:
records fall into `TypeDeclarationSyntax` case :-)